extension/encoding/avrologencodingextension/extension.go

```diff
@@ -16,26 +16,51 @@ import (
 )
 
 var (
+	_ encoding.LogsMarshalerExtension   = (*avroLogExtension)(nil)
 	_ encoding.LogsUnmarshalerExtension = (*avroLogExtension)(nil)
 )
 
 type avroLogExtension struct {
-	deserializer avroDeserializer
+	config  *Config
+	encoder avroSerDe
 }
 
 func newExtension(config *Config) (*avroLogExtension, error) {
-	deserializer, err := newAVROStaticSchemaDeserializer(config.Schema)
+	encoder, err := newAVROStaticSchemaSerDe(config.Schema)
 	if err != nil {
 		return nil, err
 	}
 
-	return &avroLogExtension{deserializer: deserializer}, nil
+	return &avroLogExtension{
+		config:  config,
+		encoder: encoder,
+	}, nil
+}
+
+func (e *avroLogExtension) MarshalLogs(logs plog.Logs) ([]byte, error) {
+	body := logs.ResourceLogs().At(0).ScopeLogs().At(0).LogRecords().At(0).Body()
+
+	var raw map[string]any
+
+	switch body.Type() {
+	case pcommon.ValueTypeMap:
+		raw = body.Map().AsRaw()
+	default:
+		return nil, fmt.Errorf("marshal: body expected to be of type 'Map' found '%v'", body.Type().String())
```

Comment:
Why is map the only type supported?

Reply:
The only possible types are map and string, as an AVRO record is always an object. In case of a string it would be required to create a JSON string within the body. Whereas for simple and small objects it should be fine, for more complex schemas with lots of fields it gets way harder to construct that in a reliable way e.g. in multiple transformation steps. So if requested we can also support it, but map gives a lot more flexibility.

Reply:
Ok